### ISSUE TITLE
no-shadowed-variable will make a mistake if there was an alone `{}` block

### ISSUE THREAD

**Issue by meowtec** [palantir/tslint#1021]
```javascript
/** case 1 */
if (true) {
    const a = 3
}
function test1() {
    const a = 5
}

/** case 2 */
{
    const b = 3
}
function test2() {
    const b = 5
}

```

Case 1 will pass, but 2 will fail. That is incorrect.
